app/controllers/report_controller/widgets.rb

```diff
@@ -655,7 +655,17 @@ def widget_set_record_vars(widget)
     #      end
     #      widget.miq_widget_shortcuts = ws
     #    end
-    widget.options[:url] = @edit[:new][:url] unless @edit[:new][:url].blank?
+    if @sb[:wtype] == "rf"
+      if @edit[:new][:feed_type] == "internal"
+        widget.resource = RssFeed.find(@edit[:new][:rss_feed_id]) if @edit[:new][:rss_feed_id]
+        widget.options.delete(:url)
+      else
+        widget.resource = nil
+        widget.options[:url] = @edit[:new][:url]
+      end
+    else
+      widget.resource = @edit[:rpt]
```

Comment:
I wonder if this logic belongs in a model.  It feels like we have 3 types of things we want to be able to switch a widget to... internal rss feed, external rss feed, or a report.  Even if we don't have 3 classes, having a single method `switch_widget_resource` that takes a resource and an optional url may make this easier to test and change in the future.  What do you think?